fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java

```diff
@@ -31,8 +31,13 @@ public class TaskRun implements Comparable<TaskRun> {
 
     private Task task;
 
-    private ConnectContext ctx;
-
+    // The ConnectCtx brought over when the TaskRun is constructed is usually the task that the user manually
+    // penalizes and executes. For the periodical task, it is triggered by the system,
+    // so typically this value may be NULL.
+    private ConnectContext buildCtx;
+    // The ConnectCtx that actually executes the SQL statement is usually copied or constructed
+    // when TaskRun is executed.
+    private ConnectContext runCtx;
```

Comment:
I think taskRun should be a background task, why it needs ConnectContext field?

Reply:
it is a bit confusing with two context.